bbq/compiler/desugar.go

```diff
@@ -58,7 +61,226 @@ func (d *Desugar) VisitVariableDeclaration(declaration *ast.VariableDeclaration)
 }
 
 func (d *Desugar) VisitFunctionDeclaration(declaration *ast.FunctionDeclaration) ast.Declaration {
-	return declaration
+	funcBlock := declaration.FunctionBlock
+	if !funcBlock.HasConditions() {
+		return declaration
+	}
+
+	statements := funcBlock.Block.Statements
+	preConditions := d.desugarConditions(funcBlock.PreConditions)
+	postConditions := d.desugarConditions(funcBlock.PostConditions)
```

Comment:
Realized we need to use the re-written post-conditions here. This is fixed in a follow-up PR